units/patch_intent.go

```diff
@@ -391,8 +392,14 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 	}
 
 	if patchDoc.IsGithubPRPatch() {
+		numCheckRuns := patchedProject.GetNumCheckRunsFromVariantTasks(patchDoc.VariantsTasks)
```

Comment:
it looks like we repeat this section a few times could we make it into a function (unless there's a reason it can't easily be done. It might not be worth it)

Reply:
numCheckRuns differs, so we can't easily extract it